stl/inc/chrono

```diff
@@ -2086,6 +2087,689 @@ namespace chrono {
+            requires _Convertable_from_utc_time<_Dest_Clock, _Duration> {
+            return _Dest_Clock::from_utc(_Time);
+        }
+    };
+
+    // [time.clock.cast.fn]
+
+    template <class _Dest_Clock, class _Source_Clock, class _Duration>
+    _NODISCARD auto clock_cast(const time_point<_Source_Clock, _Duration>& _Time) {
+        constexpr bool _Has_direct_conversion =
+            is_invocable_v<clock_time_conversion<_Dest_Clock, _Source_Clock>, decltype(_Time)>;
+
+        constexpr bool _Utc_from_src  = _Convertable_to_utc_time<_Source_Clock, _Duration>;
+        constexpr bool _Sys_from_src  = _Convertable_to_sys_time<_Source_Clock, _Duration>;
+        constexpr bool _Dest_from_utc = _Convertable_from_utc_time<_Dest_Clock, _Duration>;
+        constexpr bool _Dest_from_sys = _Convertable_from_sys_time<_Dest_Clock, _Duration>;
+
+        constexpr bool _Has_utc_conversion = _Dest_from_utc && _Utc_from_src;
+        constexpr bool _Has_sys_conversion = _Dest_from_sys && _Sys_from_src;
+        static_assert(_Has_direct_conversion || !(_Has_utc_conversion && _Has_sys_conversion),
+            "A two-step clock time conversion is required to be unique, either through utc_clock or system_clock, but "
+            "not both.");
+
+        constexpr bool _Has_sys_utc_conversion = _Dest_from_sys && _Utc_from_src;
+        constexpr bool _Has_utc_sys_conversion = _Dest_from_utc && _Sys_from_src;
+        static_assert(_Has_direct_conversion || _Has_utc_conversion || _Has_sys_conversion
+                          || !(_Has_utc_sys_conversion && _Has_sys_utc_conversion),
+            "A three-step clock time conversion is required to be unique, either utc-to-system or system-to-utc, but "
+            "not both.");
+
+        if constexpr (_Has_direct_conversion) {
+            return clock_time_conversion<_Dest_Clock, _Source_Clock>{}(_Time);
+        } else if constexpr (_Has_utc_conversion) {
+            return clock_time_conversion<_Dest_Clock, utc_clock>{}(
+                clock_time_conversion<utc_clock, _Source_Clock>{}(_Time));
+        } else if constexpr (_Has_sys_conversion) {
+            return clock_time_conversion<_Dest_Clock, system_clock>{}(
+                clock_time_conversion<system_clock, _Source_Clock>{}(_Time));
+        } else if constexpr (_Has_sys_utc_conversion) {
+            return clock_time_conversion<_Dest_Clock, system_clock>{}(clock_time_conversion<system_clock, utc_clock>{}(
+                clock_time_conversion<utc_clock, _Source_Clock>{}(_Time)));
+        } else if constexpr (_Has_utc_sys_conversion) {
+            return clock_time_conversion<_Dest_Clock, utc_clock>{}(clock_time_conversion<utc_clock, system_clock>{}(
+                clock_time_conversion<system_clock, _Source_Clock>{}(_Time)));
+        } else {
+            static_assert(!_Has_direct_conversion && !_Has_utc_conversion && !_Has_sys_conversion
+                              && !_Has_sys_utc_conversion && !_Has_utc_sys_conversion,

```

Comment:
I think `_Always_false` could simplify this, instead of effectively repeating the conditions above that have failed to be true.

Reply:
O brave new world that has such people in it.